### PR TITLE
Add early return when shadowenv returns an empty string

### PR DESCRIPTION
**Pull request by bitwise-aiden**
### Motivation

Raised by @EsterKais, shadowenv has been failing because we are expecting `shadowenv hook --json` to return valid json. It is possible that this command won't print anything to stdout which will result in an empty string instead.

### Implementation

Check command output, returning early if it is empty.

### CODE REVIEW

File: src/ruby.ts [hunk 1]
```diff
@@ -122,6 +122,11 @@ export class Ruby {
       cwd: this.workingFolder,
     });
 
+    if (result.stdout === "") {
+      // eslint-disable-next-line no-process-env
+      this._env = { ...process.env };
+      return;
+    };
```
Comment (paracycle) [inline src/ruby.ts:129]: Alternatively:
```suggestion
    if (result.stdout === "") {
      result.stdout = "{ }";
    };
```

Comment (paracycle) [reply src/ruby.ts:129]:
```js
> { ...{ foo: 1, bar: 2}, ...JSON.parse("{ }").exported }
{ foo: 1, bar: 2 }
```

File: src/ruby.ts [hunk 2]
```diff
@@ -122,6 +122,11 @@ export class Ruby {
       cwd: this.workingFolder,
     });
 
+    if (result.stdout === "") {
```
Comment (paracycle) [inline src/ruby.ts:125]: Could there be any whitespace in `stdout`? If so, we would be safer with a [`trim()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim):
```suggestion
    if (result.stdout.trim() === "") {
```

File: src/ruby.ts (hunk 1 above)
Comment (vinistock) [reply src/ruby.ts:129]: That's a neat idea.

File: src/ruby.ts (hunk 2 above)
Comment (vinistock) [reply src/ruby.ts:125]: I don't think so, but it doesn't hurt either.

**Review by EsterKais — APPROVED** [review]

**Pull request merged**